### PR TITLE
fix #21 : add marker * to the title when unsaved changes are detected.

### LINKED ISSUE #21: unsaved changes indicator in viewer
https://github.com/liam2/larray/issues/256

### PR DESCRIPTION
**Pull request by alixdamman**
+ changed behavior of var unsaved_modifications (MappingEditor class): set to True as soon as data has been changed (before users click on Apply button)

### CODE REVIEW

File: larray_editor/editor.py [hunk 1]
```diff
@@ -230,6 +230,9 @@ def void_formatter(array, *args, **kwargs):
             arrays = [k for k, v in self.data.items() if self._display_in_grid(k, v)]
             self.add_list_items(arrays)
 
+        #tracking data changes
```
Comment (gdementen) [inline larray_editor/editor.py]: s/#t/# t/

File: larray_editor/editor.py [hunk 2]
```diff
@@ -277,6 +280,23 @@ def setup_menu_bar(self):
...
+        self.unsaved_modifications = True
+
+    @property
+    def unsaved_modifications(self):
+        return self._unsaved_modifications
+
+    @unsaved_modifications.setter
+    def unsaved_modifications(self, unsaved_modifications):
+        self._unsaved_modifications = unsaved_modifications
+        # add '*' at the end of the title if unsaved modifications. Remove it if not.
```
Comment (gdementen) [inline larray_editor/editor.py]: I would rather have this setter call update_title then have update_title construct the title accordingly.

Comment (alixdamman) [reply larray_editor/editor.py]: This is what I implemented first but I thought it was an overkill. 
I guess update_title does not take so much computation time.

**Comment by alixdamman** [comment]
updated and added a commit to fix a bug in Adapter.set_data()

**Pull request merged**